torch/distributed/device_mesh.py

```diff
@@ -254,9 +254,13 @@ def _get_or_create_default_group(self):
             return _get_default_group()
 
         def _init_process_groups(self):
-            # group tag/ranks associated with each mesh dimension, each mesh dimension should
-            # have one sub-group per rank
-            dim_group_infos: List[Tuple[str, List[int]]] = []
+            # tag/ranks/group_name associated with each mesh dimension, each
+            # mesh dimension should have one sub-group per rank
+            #
+            # TODO(yifu): remove tag and ranks once we fully migrate to native
+            # functional collectives. See details in:
+            # https://github.com/pytorch/pytorch/issues/93173#issuecomment-1907095208
+            dim_group_infos: List[Tuple[str, List[int], str]] = []
```

Comment:
quick comment: would `tag` have the same semantic with `group_name`? i.e. are they the same str? If they are maybe we can reuse the tag?

Reply:
They are different things. tag isn't always present so it's tricky to reuse.